src/plugins/intel_npu/src/backend/src/zero_infer_request.cpp

```diff
@@ -92,11 +105,7 @@ std::optional<size_t> determine_dynamic_batch_size(const IODescriptor& desc,
         return std::nullopt;
     }
 
-    if ((*desc.shapeFromIRModel)[intel_npu::utils::BATCH_AXIS].is_dynamic()) {
-        return tensor->get_shape()[intel_npu::utils::BATCH_AXIS];
-    }
-
-    return std::nullopt;
+    return tensor->get_shape()[intel_npu::utils::BATCH_AXIS];
```

Comment:
If `wasDynamic == true` is equivalent to `origShape[intel_npu::utils::BATCH_AXIS].is_dynamic()`